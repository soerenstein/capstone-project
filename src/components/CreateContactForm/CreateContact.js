import React from 'react'
import CreateContactForm from './CreateContactForm'
import Header from '../Header/Header'

export default function CreateContact({ onSubmit }) {
    return (
        <div>
            <Header />
            <CreateContactForm />
        </div>
    )
}
