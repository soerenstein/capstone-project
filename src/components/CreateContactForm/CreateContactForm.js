import React, { useState } from 'react'
import styled from 'styled-components/macro'
import { v4 as uuidv4 } from 'uuid'
import CardWrapper from '../Design/CardWrapper/CardWrapper'
import PageLayout from '../Design/CardWrapper/PageLayout'

export default function CreateContactForm({ onSubmit }) {
  const [firstName, setFirstName] = useState(false)
  const [lastName, setLastName] = useState(false)

  const disabledButton = !firstName && !lastName

  return (
    <PageLayout title="Kontakt anlegen">
      <CardWrapper>
        <FormStyled onSubmit={handleSubmit}>
          <LabelWrapper>
            <LabelStyled htmlFor="firstName">
              <LabelName>Vorname</LabelName>
              <InputStyled
                autoFocus
                name="firstName"
                id="firstName"
                type="text"
                maxLength="40"
                onChange={handleChange}
                placeholder="Max"
              />
            </LabelStyled>
            <LabelStyled htmlFor="lastName">
              <LabelName>Nachname</LabelName>

              <InputStyled
                name="lastName"
                id="lastName"
                type="text"
                maxLength="40"
                onChange={handleChange}
                placeholder="Mustermann"
              />
            </LabelStyled>
          </LabelWrapper>
          <LabelStyled htmlFor="company">
            <LabelName>Unternehmen</LabelName>

            <InputStyled
              name="company"
              id="company"
              type="text"
              placeholder="Universal GmbH"
            />
          </LabelStyled>
          <LabelStyled htmlFor="phone">
            <LabelName>Telefon</LabelName>
            <InputStyled
              name="phone"
              type="tel"
              id="phone"
              placeholder="0176-12345678"
            />
          </LabelStyled>
          <LabelStyled htmlFor="mail">
            <LabelName>E-Mail</LabelName>
            <InputStyled name="mail" id="mail" placeholder="hello@me.de" />
          </LabelStyled>
          <LabelStyled htmlFor="street">
            <LabelName>Adresse</LabelName>
            <InputStyled
              name="zip"
              type="text"
              id="street"
              placeholder="Straße"
            />
            <LabelWrapper>
              <InputStyled name="zip" type="text" id="zip" placeholder="PLZ" />
              <InputStyled
                name="city"
                type="text"
                id="city"
                placeholder="Stadt"
              />
            </LabelWrapper>
          </LabelStyled>
          <LabelStyled htmlFor="note">
            <LabelName>Notiz</LabelName>

            <InputStyled name="note" type="text" id="note" />
          </LabelStyled>
          <LabelStyled htmlFor="category">
            <LabelName>Kategorie</LabelName>

            <InputStyled
              name="category"
              type="text"
              id="category"
              placeholder="Kunde, Dienstleister"
            />
          </LabelStyled>
          <ButtonStyled id="submit" disabled={disabledButton}>
            Kontakt hinzufügen
          </ButtonStyled>
        </FormStyled>
      </CardWrapper>
    </PageLayout>
  )

  function handleChange(event) {
    event.target.id === 'firstName' &&
      setFirstName(
        event.target.id === 'firstName' && event.target.value !== ''
          ? true
          : false
      )
    event.target.id === 'lastName' &&
      setLastName(
        event.target.id === 'lastName' && event.target.value !== ''
          ? true
          : false
      )
  }

  function handleSubmit(event) {
    event.preventDefault()
    const form = event.target
    const contactItem = {
      firstName: form.firstName.value,
      lastName: form.lastName.value,
      company: form.company.value,
      phone: form.phone.value,
      mail: form.mail.value,
      street: form.street.value,
      zip: form.zip.value,
      city: form.city.value,
      note: form.note.value,
      category: form.category.value,
      id: uuidv4(),
    }
    onSubmit(contactItem)
    form.reset()
    form[0] && form[0].focus()
  }
}

const FormStyled = styled.form`
  display: grid;
  gap: 10px;
  margin: auto;
  margin-top: 40px;
`
const LabelWrapper = styled.div`
  display: grid;
  grid-template-columns: auto auto;
  gap: 10px;
`

const LabelStyled = styled.label`
  display: grid;
  color: #ffffff;
`

const LabelName = styled.div`
  padding-left: 20px;
`
const InputStyled = styled.input`
  border-style: none;
  padding-left: 20px;
  height: 50px;
  box-shadow: 0px 5px 10px #00000029;
  border-radius: 30px;
  outline: none;
  margin-bottom: 15px;
`

const ButtonStyled = styled.button`
  padding: 17px 48px;
  font-size: 20px;
  background: #ffffff 0% 0% no-repeat padding-box;
  box-shadow: 0px 5px 10px #00000029;
  border-style: none;
  border-radius: 30px;
  opacity: 1;
  margin: 20px 0 30px 0;

  &:disabled {
    opacity: 0.5;
  }
`
