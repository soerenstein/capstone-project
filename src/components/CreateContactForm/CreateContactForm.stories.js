import React from 'react'
import CreateContactForm from './CreateContactForm'

export default {
  title: 'CreateContactForm',
  component: CreateContactForm,
}

export const Form = () => <CreateContactForm />
