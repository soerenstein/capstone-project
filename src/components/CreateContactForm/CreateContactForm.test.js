import React from 'react'
import { render } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import CreateContactForm from './CreateContactForm'
import { MemoryRouter } from 'react-router-dom'

describe('The form', () => {
  it('renders correctly', () => {
    const tree = render(
      <MemoryRouter>
        <CreateContactForm />
      </MemoryRouter>
    )
    expect(tree).toMatchSnapshot()
  })
})

describe('The labels', () => {
  it(' renders correctly', () => {
    const { getByText } = render(
      <MemoryRouter>
        <CreateContactForm />
      </MemoryRouter>
    )
    expect(getByText('Vorname')).toBeInTheDocument()
    expect(getByText('Nachname')).toBeInTheDocument()
    expect(getByText('Unternehmen')).toBeInTheDocument()
    expect(getByText('Telefon')).toBeInTheDocument()
    expect(getByText('E-Mail')).toBeInTheDocument()
    expect(getByText('Adresse')).toBeInTheDocument()
    expect(getByText('Notiz')).toBeInTheDocument()
    expect(getByText('Kategorie')).toBeInTheDocument()
  })
})
