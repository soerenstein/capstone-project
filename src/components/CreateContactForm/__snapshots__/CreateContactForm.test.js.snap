// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The form renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <main
        class="PageLayout__PageStyled-i5mqh4-0 jxwuTH"
      >
        <div
          class="NavBar__HeaderWrapper-jwnntt-0 iTVKYa"
        >
          <a
            aria-current="page"
            class="active"
            href="/"
          >
            <svg
              class="NavBar__ArrowStyled-jwnntt-1 mugrW"
              id="back"
            >
              arrow-left.svg
            </svg>
          </a>
          <a
            href="/favorites"
          >
            <svg
              class="NavBar__FavoriteStyled-jwnntt-2 hjlejN"
            >
              heart.svg
            </svg>
          </a>
          <a
            href="/user-profile"
          >
            <svg
              class="NavBar__ProfileStyled-jwnntt-3 bLkKnR"
            >
              user-header.svg
            </svg>
          </a>
          <a
            href="/create"
          >
            <svg
              class="NavBar__CreateStyled-jwnntt-4 ihxMvS"
            >
              user-plus.svg
            </svg>
          </a>
        </div>
        <h1
          class="PageLayout__PageName-i5mqh4-1 fPelXI"
        >
          Kontakt anlegen
        </h1>
        <div
          class="CardWrapper__Shadow-m1w3w7-1 hUnpQ"
        />
        <div
          class="CardWrapper__StyledCardWrapper-m1w3w7-0 dXOIHJ"
        >
          <button
            class="CreateContactForm__QRButtonStyled-sc-1xbnsj5-6 dVLnrA"
          >
            QR-Code scannen
          </button>
          <form
            class="CreateContactForm__FormStyled-sc-1xbnsj5-0 hELOHk"
          >
            <div
              class="CreateContactForm__LabelWrapper-sc-1xbnsj5-1 bSEKSw"
            >
              <label
                class="CreateContactForm__LabelStyled-sc-1xbnsj5-2 ldDYpC"
                for="firstName"
              >
                <div
                  class="CreateContactForm__LabelName-sc-1xbnsj5-3 ewbFFX"
                >
                  Vorname
                </div>
                <input
                  class="CreateContactForm__InputStyled-sc-1xbnsj5-4 djFcGd"
                  id="firstName"
                  maxlength="40"
                  name="firstName"
                  placeholder="Max"
                  required=""
                  type="text"
                />
              </label>
              <label
                class="CreateContactForm__LabelStyled-sc-1xbnsj5-2 ldDYpC"
                for="lastName"
              >
                <div
                  class="CreateContactForm__LabelName-sc-1xbnsj5-3 ewbFFX"
                >
                  Nachname
                </div>
                <input
                  class="CreateContactForm__InputStyled-sc-1xbnsj5-4 djFcGd"
                  id="lastName"
                  maxlength="40"
                  name="lastName"
                  placeholder="Mustermann"
                  required=""
                  type="text"
                />
              </label>
            </div>
            <label
              class="CreateContactForm__LabelStyled-sc-1xbnsj5-2 ldDYpC"
              for="company"
            >
              <div
                class="CreateContactForm__LabelName-sc-1xbnsj5-3 ewbFFX"
              >
                Unternehmen
              </div>
              <input
                class="CreateContactForm__InputStyled-sc-1xbnsj5-4 djFcGd"
                id="company"
                name="company"
                placeholder="Universal GmbH"
                required=""
                type="text"
              />
            </label>
            <label
              class="CreateContactForm__LabelStyled-sc-1xbnsj5-2 ldDYpC"
              for="phone"
            >
              <div
                class="CreateContactForm__LabelName-sc-1xbnsj5-3 ewbFFX"
              >
                Telefon
              </div>
              <input
                class="CreateContactForm__InputStyled-sc-1xbnsj5-4 djFcGd"
                id="phone"
                name="phone"
                placeholder="0176-12345678"
                type="tel"
              />
            </label>
            <label
              class="CreateContactForm__LabelStyled-sc-1xbnsj5-2 ldDYpC"
              for="mail"
            >
              <div
                class="CreateContactForm__LabelName-sc-1xbnsj5-3 ewbFFX"
              >
                E-Mail
              </div>
              <input
                class="CreateContactForm__InputStyled-sc-1xbnsj5-4 djFcGd"
                id="mail"
                name="mail"
                placeholder="hello@me.de"
                type="email"
              />
            </label>
            <label
              class="CreateContactForm__LabelStyled-sc-1xbnsj5-2 ldDYpC"
              for="street"
            >
              <div
                class="CreateContactForm__LabelName-sc-1xbnsj5-3 ewbFFX"
              >
                Adresse
              </div>
              <input
                class="CreateContactForm__InputStyled-sc-1xbnsj5-4 djFcGd"
                id="street"
                name="street"
                placeholder="Straße"
                type="text"
              />
              <div
                class="CreateContactForm__LabelWrapper-sc-1xbnsj5-1 bSEKSw"
              >
                <input
                  class="CreateContactForm__InputStyled-sc-1xbnsj5-4 djFcGd"
                  id="zip"
                  name="zip"
                  placeholder="PLZ"
                  type="text"
                />
                <input
                  class="CreateContactForm__InputStyled-sc-1xbnsj5-4 djFcGd"
                  id="city"
                  name="city"
                  placeholder="Stadt"
                  type="text"
                />
              </div>
            </label>
            <label
              class="CreateContactForm__LabelStyled-sc-1xbnsj5-2 ldDYpC"
              for="note"
            >
              <div
                class="CreateContactForm__LabelName-sc-1xbnsj5-3 ewbFFX"
              >
                Notiz
              </div>
              <input
                class="CreateContactForm__InputStyled-sc-1xbnsj5-4 djFcGd"
                id="note"
                name="note"
                type="text"
              />
            </label>
            <label
              class="CreateContactForm__LabelStyled-sc-1xbnsj5-2 ldDYpC"
              for="category"
            >
              <div
                class="CreateContactForm__LabelName-sc-1xbnsj5-3 ewbFFX"
              >
                Kategorie
              </div>
              <input
                class="CreateContactForm__InputStyled-sc-1xbnsj5-4 djFcGd"
                id="category"
                name="category"
                placeholder="Kunde, Dienstleister"
                type="text"
              />
            </label>
            <button
              class="CreateContactForm__ButtonStyled-sc-1xbnsj5-5 eATByy"
              disabled=""
              id="submit"
            >
              Kontakt hinzufügen
            </button>
          </form>
        </div>
      </main>
    </div>
  </body>,
  "container": <div>
    <main
      class="PageLayout__PageStyled-i5mqh4-0 jxwuTH"
    >
      <div
        class="NavBar__HeaderWrapper-jwnntt-0 iTVKYa"
      >
        <a
          aria-current="page"
          class="active"
          href="/"
        >
          <svg
            class="NavBar__ArrowStyled-jwnntt-1 mugrW"
            id="back"
          >
            arrow-left.svg
          </svg>
        </a>
        <a
          href="/favorites"
        >
          <svg
            class="NavBar__FavoriteStyled-jwnntt-2 hjlejN"
          >
            heart.svg
          </svg>
        </a>
        <a
          href="/user-profile"
        >
          <svg
            class="NavBar__ProfileStyled-jwnntt-3 bLkKnR"
          >
            user-header.svg
          </svg>
        </a>
        <a
          href="/create"
        >
          <svg
            class="NavBar__CreateStyled-jwnntt-4 ihxMvS"
          >
            user-plus.svg
          </svg>
        </a>
      </div>
      <h1
        class="PageLayout__PageName-i5mqh4-1 fPelXI"
      >
        Kontakt anlegen
      </h1>
      <div
        class="CardWrapper__Shadow-m1w3w7-1 hUnpQ"
      />
      <div
        class="CardWrapper__StyledCardWrapper-m1w3w7-0 dXOIHJ"
      >
        <button
          class="CreateContactForm__QRButtonStyled-sc-1xbnsj5-6 dVLnrA"
        >
          QR-Code scannen
        </button>
        <form
          class="CreateContactForm__FormStyled-sc-1xbnsj5-0 hELOHk"
        >
          <div
            class="CreateContactForm__LabelWrapper-sc-1xbnsj5-1 bSEKSw"
          >
            <label
              class="CreateContactForm__LabelStyled-sc-1xbnsj5-2 ldDYpC"
              for="firstName"
            >
              <div
                class="CreateContactForm__LabelName-sc-1xbnsj5-3 ewbFFX"
              >
                Vorname
              </div>
              <input
                class="CreateContactForm__InputStyled-sc-1xbnsj5-4 djFcGd"
                id="firstName"
                maxlength="40"
                name="firstName"
                placeholder="Max"
                required=""
                type="text"
              />
            </label>
            <label
              class="CreateContactForm__LabelStyled-sc-1xbnsj5-2 ldDYpC"
              for="lastName"
            >
              <div
                class="CreateContactForm__LabelName-sc-1xbnsj5-3 ewbFFX"
              >
                Nachname
              </div>
              <input
                class="CreateContactForm__InputStyled-sc-1xbnsj5-4 djFcGd"
                id="lastName"
                maxlength="40"
                name="lastName"
                placeholder="Mustermann"
                required=""
                type="text"
              />
            </label>
          </div>
          <label
            class="CreateContactForm__LabelStyled-sc-1xbnsj5-2 ldDYpC"
            for="company"
          >
            <div
              class="CreateContactForm__LabelName-sc-1xbnsj5-3 ewbFFX"
            >
              Unternehmen
            </div>
            <input
              class="CreateContactForm__InputStyled-sc-1xbnsj5-4 djFcGd"
              id="company"
              name="company"
              placeholder="Universal GmbH"
              required=""
              type="text"
            />
          </label>
          <label
            class="CreateContactForm__LabelStyled-sc-1xbnsj5-2 ldDYpC"
            for="phone"
          >
            <div
              class="CreateContactForm__LabelName-sc-1xbnsj5-3 ewbFFX"
            >
              Telefon
            </div>
            <input
              class="CreateContactForm__InputStyled-sc-1xbnsj5-4 djFcGd"
              id="phone"
              name="phone"
              placeholder="0176-12345678"
              type="tel"
            />
          </label>
          <label
            class="CreateContactForm__LabelStyled-sc-1xbnsj5-2 ldDYpC"
            for="mail"
          >
            <div
              class="CreateContactForm__LabelName-sc-1xbnsj5-3 ewbFFX"
            >
              E-Mail
            </div>
            <input
              class="CreateContactForm__InputStyled-sc-1xbnsj5-4 djFcGd"
              id="mail"
              name="mail"
              placeholder="hello@me.de"
              type="email"
            />
          </label>
          <label
            class="CreateContactForm__LabelStyled-sc-1xbnsj5-2 ldDYpC"
            for="street"
          >
            <div
              class="CreateContactForm__LabelName-sc-1xbnsj5-3 ewbFFX"
            >
              Adresse
            </div>
            <input
              class="CreateContactForm__InputStyled-sc-1xbnsj5-4 djFcGd"
              id="street"
              name="street"
              placeholder="Straße"
              type="text"
            />
            <div
              class="CreateContactForm__LabelWrapper-sc-1xbnsj5-1 bSEKSw"
            >
              <input
                class="CreateContactForm__InputStyled-sc-1xbnsj5-4 djFcGd"
                id="zip"
                name="zip"
                placeholder="PLZ"
                type="text"
              />
              <input
                class="CreateContactForm__InputStyled-sc-1xbnsj5-4 djFcGd"
                id="city"
                name="city"
                placeholder="Stadt"
                type="text"
              />
            </div>
          </label>
          <label
            class="CreateContactForm__LabelStyled-sc-1xbnsj5-2 ldDYpC"
            for="note"
          >
            <div
              class="CreateContactForm__LabelName-sc-1xbnsj5-3 ewbFFX"
            >
              Notiz
            </div>
            <input
              class="CreateContactForm__InputStyled-sc-1xbnsj5-4 djFcGd"
              id="note"
              name="note"
              type="text"
            />
          </label>
          <label
            class="CreateContactForm__LabelStyled-sc-1xbnsj5-2 ldDYpC"
            for="category"
          >
            <div
              class="CreateContactForm__LabelName-sc-1xbnsj5-3 ewbFFX"
            >
              Kategorie
            </div>
            <input
              class="CreateContactForm__InputStyled-sc-1xbnsj5-4 djFcGd"
              id="category"
              name="category"
              placeholder="Kunde, Dienstleister"
              type="text"
            />
          </label>
          <button
            class="CreateContactForm__ButtonStyled-sc-1xbnsj5-5 eATByy"
            disabled=""
            id="submit"
          >
            Kontakt hinzufügen
          </button>
        </form>
      </div>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
